app/code/Magento/Store/Model/StoreCreateManagement.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Magento\Store\Model;
+
+use Magento\Store\Api\StoreCreateManagementInterface;
+
+class StoreCreateManagement implements StoreCreateManagementInterface
+{
+    /**
+     * @var \Magento\Store\Model\WebsiteFactory
+     */
+    private $websiteFactory;
+
+    /**
+     * @var \Magento\Store\Model\GroupFactory
+     */
+    private $groupFactory;
+
+    /**
+     * @var \Magento\Store\Model\StoreFactory
+     */
+    private $storeFactory;
+
+    /**
+     *
+     * @param \Magento\Store\Model\WebsiteFactory     $websiteFactory
+     * @param \Magento\Store\Model\GroupFactory       $groupFactory
+     * @param \Magento\Store\Model\StoreFactory       $storeFactory
+     */
+    public function __construct(
+        \Magento\Store\Model\WebsiteFactory $websiteFactory,
+        \Magento\Store\Model\GroupFactory $groupFactory,
+        \Magento\Store\Model\StoreFactory $storeFactory
+    ) {
+        $this->websiteFactory = $websiteFactory;
+        $this->groupFactory = $groupFactory;
+        $this->storeFactory = $storeFactory;
+    }
+
+    /**
+     *
+     * @param  array $data
+     * @return \Magento\Store\Model\Store
+     */
+    public function create($data)
+    {
+        /** @var \Magento\Store\Model\Store $storeModel */
+        $storeModel = $this->storeFactory->create();
+        $storeModel->setData($data);
```

Comment:
`setData` is not an interface method, however the data can be passed directly to the factory
```suggestion
        $storeModel = $this->storeFactory->create(['data' => $data]);
```